### PR TITLE
Fix authentication error when creating subscription second time with valid credentials

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Resolves https://github.com/crate/crate/issues/12462

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
retest this please

**Comment by BaurzhanSakhariev** [comment]
added `@Repeat(iterations = 20)` and tried ti reproduce locally with `./gradlew :server:test -Dtests.seed=2797EF175DC5DF8A -Dtests.class=io.crate.integrationtests.LogicalReplicationITest -Dtests.method="test_write_to_subscribed_empty_partitioned_table_is_forbidden" -Dtests.locale=fr-FR -Dtests.timezone=Africa/Nouakchott` but it passed locally

retest this please

File: server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java [hunk 1]
```diff
@@ -180,7 +180,7 @@ public void test_pg_client_can_use_password_authentication_for_replication() thr
     }
 
     @Test
-    public void test_cannot_create_subscription_with_invalid_user_password() throws Exception {
+    public void test_cannot_create_subscription_with_invalid_user_password_and_can_create_with_same_name_and_valid_credentials() throws Exception {
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java:183]: Can we create a `RemoterClustersTest` unit test instead?

Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java]: done in https://github.com/crate/crate/pull/12507/commits/15c05ab199b54c27ec65f80d2e2e0c55d68c6053

**Review by seut — APPROVED** [review]
👍

**Comment by BaurzhanSakhariev** [comment]
no test failures, from logs:
```
Still waiting to schedule task
Waiting for next available executor on 
cancelling nested steps due to timeout
```

retest this please

**Comment by BaurzhanSakhariev** [comment]
infrastructural issues, `* is reserved for jobs with matching label expression;` - looks like appropriate machines are busy?
retest this please

**Comment by BaurzhanSakhariev** [comment]
retest this please

**Comment by BaurzhanSakhariev** [comment]
retest this please

**Pull request merged**